### PR TITLE
Unique connector webview / Iframe

### PR DESCRIPTION
**Pull request by GrandSchtroumpf**

### CODE REVIEW

File: projects/client-vscode/src/webview.ts [hunk 1]
```diff
@@ -1,51 +1,76 @@
...
+  PluginOptions,
+  checkOrigin
+} from '@remixproject/plugin'
 
 declare const acquireVsCodeApi: any
-
 /**
  * This Webview connector
  */
 export class WebviewConnector implements ClientConnector {
```
Comment (yann300) [inline projects/client-vscode/src/webview.ts:19]: shouldn't this be in a `connector.ts` (like for iframe.ts) or is this fine to keep it here?

File: projects/client-vscode/src/webview.ts [hunk 2]
```diff
@@ -1,51 +1,76 @@
...
 
+
   /** Send a message to the engine */
   send(message: Partial<Message>) {
-    this.postMessage(message)
+    if (this.isVscode) {
+      this.source.postMessage(message)
+    } else if (this.origin || isHandshake(message)) {
+      const origin = this.origin || '*'
+      this.source.postMessage(message, origin)
```
Comment (yann300) [inline projects/client-vscode/src/webview.ts:36]: don't need to use `window.parent.` anymore ? like for https://github.com/ethereum/remix-plugin/blob/v0.3.0/projects/client-iframe/src/connector.ts#L21

File: webpack.config.js [hunk 3]
```diff
@@ -133,6 +133,9 @@ const wsClient = {
 // Iframe client
 const iframeClient = {
   ...config('client-iframe'),
+  externals: {
+    '@remixproject/plugin': 'commonjs2 @remixproject/plugin',
```
Comment (yann300) [inline webpack.config.js:137]: is this file packaged `commonjs2` ?

File: projects/client-vscode/src/webview.ts (hunk 2 above)
Comment (GrandSchtroumpf) [reply projects/client-vscode/src/webview.ts:36]: It's the default source when parent s not vscode. : line 26
`this.source = this.isVscode ? acquireVsCodeApi() : window.parent`

File: projects/client-vscode/src/webview.ts (hunk 1 above)
Comment (GrandSchtroumpf) [reply projects/client-vscode/src/webview.ts:19]: I think `connector.ts` in iframe should be renamed. But I'm still figuring out how to organize this repo. I' happy to update, but I would prefer to se how it comes out.

**Pull request merged**